### PR TITLE
Fix Safari 9

### PR DESCRIPTION
**Pull request by msokk**
Removing the `data:` part from blob url makes it load again in a new tab.
Haven't tested how it still works in Safari 8 though.

### CODE REVIEW

**Comment by rndme** [comment]
thanks! i'll do some back testing, but the change makes sense. i'll try to get it in soon if everything checks out.

**Comment by msokk** [comment]
Any updates?

**Comment by rndme** [comment]
i've been on vacation away from my mac and busy with family. you picked a bad time of year, the worst in fact... sorry it's taking so long; i haven't forgotten and i should be able to check it out pretty soon.

**Comment by msokk** [comment]
Thats ok :) Take your time

**Comment by rndme** [comment]
`if(typeof safari !=="undefined" ){ ` seems to be false in safari9, so the change won't actually affect that, but could break safari8, which kinda worked the last time i checked, but i can no longer test it out. i'll look into the whole situation more, maybe somebody out there has found a solution we can incorporate. 

thanks for the heads up and the code, sorry it doesn't fix safari (the new IE?); i was excited...

**Comment by msokk** [comment]
Hi, weird, library with this patch is running live on: https://my.teleport.org/runway/

Also my Safari 9 console output:
![Screenshot]
(https://photos-1.dropbox.com/t/2/AABAPixkuBEM6jX3i3nu-S_2JnZYsPzIVjPz0aZGtbz-9Q/12/463359/png/32x32/3/1452438000/0/2/Screenshot%202016-01-10%2012.24.12.png/EJOnRRimwbb1AyACKAI/Wxouc3XMDdIdlNISyUrQAWE6wItCTuQP29e5CELX4aw?size_mode=5&size=32x32)

**Comment by msokk** [comment]
Booted up Yosemite on Virtualbox, works on Safari 8 too - [Screenshot](https://www.dropbox.com/s/ne80nj6pmdizhdr/Screenshot%202016-01-10%2017.00.56.png?dl=0)

Of course in a fallback mode, by opening a plain text blob in new tab.

**Comment by msokk** [comment]
Also `window.safari` is the same on vanilla Safari 8.

**Comment by Spy-Seth** [comment]
I'm facing the same issue. The patch proposed by @msokk works perfectly on Safari 9. By playing around I find that we could also redirect to the blob URL on this browser:

```javascript
var blobUrl = window.URL.createObjectURL(myBlob);
window.location = blobUrl;
```

**Comment by rndme** [comment]
what am i missing in my testing? it didn't do anything for me when i tried. it seems the patch cannot affect safari9 because safari9 will never make it to line 110 because safari9 will not satisfy the conditional on line 109: `if(window.safari)`, which stopped being set in safari8. therefore, if the patch works on safari9, then so too should the code without the patch, but it doesn't not. msokk showed a shot of the userAgent, but since safari allows changing that, we can't reliably use it. 

that said, if a blob routine is now workable in safari, then it might be possible to fix somehow, it's just that this patch changes a part of the code safari9 can't see.

thanks for the feedback and info about URL, i will definitely check this out and see if it can be used to fix safari.

oh, and if i am misreading the patch somehow, i apologize in advance; i tried it without improvement and then realized that `window.safari` is not set in safari9, so i stopped testing/debugging; be insistent if that was in haste...

**Comment by rndme** [comment]
i would really like to get this fixed, but i don't want to just apply a patch that didn't work for me in testing...

the test page i'm using is at http://pagedemos.com/nqcrj2uy6ku4/ if you want to kick the tires...

**Comment by msokk** [comment]
@rndme I found out why window.safari is undefined for you.
Given that it contains only `pushNotification` property, some Safari security system for this disables the whole `window.safari` property, when accessed in an iframe (pushnotifications should not be registered in a frame)
and your demo link is in an `<iframe>`.

So I agree, we should have a better detection of Safari.

**Comment by msokk** [comment]
Updated the pull request with [Safari user agent test](https://github.com/ua-parser/uap-core/blob/master/regexes.yaml#L472) from [uap-core](https://github.com/ua-parser/uap-core) project.

**Comment by rndme** [comment]
kudos on solving the iframe mystery, that explains a lot. i'm not in love with sniffing userAgent strings, and i suppose an iframe in an iframe is a edge case. let me test again without an iframe in play see what results. i might have you revert to the `window.safari` version and merge that. 

tldr; thanks, one final round of testing and i'll get it pulled...

**Comment by msokk** [comment]
I agree with UA sniffing being unreliable, I tried to use as tested source for that regexp as I could, uap-core is used by user-agent package and cdn.polyfill.io

If we're reverting back to window.safari, there should be a disclaimer in README.

**Comment by Spy-Seth** [comment]
:+1: 

Could you release a new version with this fix?

**Comment by rndme** [comment]
done. must have slipped my mind last night, thanks for the heads-up!
